pcdsutils/enum.py

```diff
@@ -0,0 +1,97 @@
+from __future__ import annotations
+
+import enum
+from typing import Iterator, Union
+
+EnumId = Union[enum.Enum, int, str]
+
+
+class CaseInsensitiveEnumMeta(enum.EnumMeta):
+    def __getattr__(self, key: EnumId) -> enum.Enum:
+        if hasattr(key, "lower"):
+            for item in self:
+                if item.name.lower() == key.lower():
+                    return item
+        return super().__getattr__(key)
+
+    def __getitem__(self, key: EnumId) -> enum.Enum:
+        if hasattr(key, "lower"):
+            for item in self:
+                if item.name.lower() == key.lower():
+                    return item
+        return super().__getitem__(key)
+
+
+class HelpfulIntEnum(enum.IntEnum, metaclass=CaseInsensitiveEnumMeta):
+    """
+    IntEnum subclass with some utility extensions and case insensitivity.
+    """
+
+    @classmethod
+    def from_any(cls, identifier: EnumId) -> HelpfulIntEnum:
+        """
+        Try all the ways to interpret identifier as the enum.
+        This is intended to consolidate the try/except tree typically used
+        to interpret external input as an enum.
+
+        Parameters
+        ----------
+        identifier : EnumId
+            Any str, int, or Enum value that corresponds with a valid value
+            on this HelpfulIntEnum instance.
+
+        Returns
+        -------
+        enum : HelpfulIntEnum
+            The corresponding enum object associated with the identifier.
+        """
```

Comment:
To document the `ValueError`:
```suggestion
        Raises
        ------
        ValueError
            If the identifier cannot be matched to any enum defined here.
        """
```
Based on https://numpydoc.readthedocs.io/en/latest/format.html#raises

Reply:
Ack, sorry - I had something like that and forgot to push that commit. That description look OK? I don't mind switching it up to this version.

Reply:
Yeah looks good